workflow/workflow.go

```diff
@@ -692,3 +707,12 @@ func DeterministicKeys[K constraints.Ordered, V any](m map[K]V) []K {
 func DeterministicKeysFunc[K comparable, V any](m map[K]V, cmp func(K, K) int) []K {
 	return internal.DeterministicKeysFunc(m, cmp)
 }
+
+// AllHandlersFinished returns true if all update handlers have finished execution.
+// Consider waiting on this condition before workflow return or continue-as-new, to prevent
+// interruption of in-progress handlers by workflow exit:
+//
+//	workflow.Await(ctx, () -> workflow.AllHandlersFinished(ctx))
```

Comment:
```suggestion
//	workflow.Await(ctx, func() bool { return workflow.AllHandlersFinished(ctx) })
```

A bit more accurate, but doesn't matter much. May also want to update the warning message that has `await` statements and such which aren't a thing in Go.